File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/model/domain/ResearcherProfileMarks.java
package com.ibcs.idsdp.rpm.model.domain;

import com.ibcs.idsdp.common.model.domain.BaseEntity;
import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Table(name = "m1_researcher_profile_marks")
@Entity
@Data
public class ResearcherProfileMarks extends BaseEntity {

	@ManyToOne
	@JoinColumn(name = "researcher_proposal_id", nullable = false, unique = true)
	private ResearcherProposal researcherProposal;

	@NotNull
	private Long stResearchCatTypeId;

	private Integer applicantAgeGovtEmployeeMarks;

	private Integer professionMarks;

	private Integer subjectRelevancyMarks;

	private Integer educationQualificationMarks;

	private String graduationType;

	private Double graduationMarks;

	private String postGraduationType;

	private Double postGraduationMarks;

	private Integer mphilPhdMarks;

	private Integer mphilMarks;

	private Integer phdMarks;

	private Integer thesisGroupMarks;

	private Integer nonThesisGroupMarks;

	private Integer thesisMarks;

	private Integer nonThesisMarks;

	private Integer publicationMarks;

	private Integer localPublicationMarks;

	private Integer internalPublicationMarks;

	private Integer researchTrainingMarks;

	private Integer researchWorkMarks;

//	private Integer thesisOnPostGraduationMarks;

	private Integer graduationRelatedToSociologyMarks;

	private Integer institutionalTypeInvolvementMarks;

	private Integer institutionalFunctionalInvolvementMarks;


	private String realAgeGovtEmployee;

	private String realProfession;

	private String realSubjectRelevancy;

	private String realEducationQualification;

	private String realGraduation;

	private String realPostGraduation;

	private String realMphilPhd;

	private String realMphil;

	private String realPhd;

	private String realThesisGroup;

	private String realThesis;

	private String realNonThesis;

	private String realNonThesisGroup;

	private String realPublication;

	private String realLocalPublication;

	private String realInternalPublication;

	private String realResearchTraining;

	private String realResearchWork;

//	private String realThesisOnPostGraduation;

	private String realGraduationRelatedToSociology;

	private String realInstitutionalTypeInvolvement;

	private String realInstitutionalFunctionalInvolvement;

	private Integer totalMarks;

}

File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/models/ResearcherProfileMarks.ts
export class ResearcherProfileMarks {

    id: number;
    uuid: string;
    researcherProposalId: number;
    stResearchCatTypeId: number;
    applicantAgeGovtEmployeeMarks: number;
    professionMarks: number;
    subjectRelevancyMarks: number;
    educationQualificationMarks: number;
    graduationType: string;
    graduationMarks: number;
    postGraduationType: string;
    postGraduationMarks: number;
    mphilPhdMarks: number;
    mphilMarks: number;
    phdMarks: number;
    thesisGroupMarks: number;
    nonThesisGroupMarks: number;
    nonThesisMarks: number;
    thesisMarks: number;
    publicationMarks: number;
    localPublicationMarks: number;
    internalPublicationMarks: number;
    researchWorkMarks: number;
    researchTrainingMarks: number;
    // thesisOnPostGraduationMarks: number;
    graduationRelatedToSociologyMarks: number;
    institutionalTypeInvolvementMarks: number = 0;
    institutionalFunctionalInvolvementMarks: number = 0;

    realAgeGovtEmployee: string;
    realProfession: string;
    realSubjectRelevancy: string;
    realEducationQualification: string;
    realGraduation: string;
    realPostGraduation: string;
    realMphilPhd: string;
    realMphil: string;
    realPhd: string;
    realThesisGroup: string;
    realThesis: string;
    realNonThesis: string;
    realNonThesisGroup: string;
    realPublication: string;
    realLocalPublication: string;
    realInternalPublication: string;
    realResearchWork: any;
    realResearchTraining: string;
    // realThesisOnPostGraduation: string;
    realGraduationRelatedToSociology: string;
    realInstitutionalTypeInvolvement: string;
    realInstitutionalFunctionalInvolvement: string;

    totalMarks: number;
    researcherProfilePersonalInfoDto: any;

    isGovEmployee: any;

}
